static/src/javascripts/projects/commercial/modules/hosted/onward.js

```diff
@@ -1,54 +1,61 @@
 // @flow
 import config from 'lib/config';
-import mediator from 'lib/mediator';
 import fetchJson from 'lib/fetch-json';
 import fastdom from 'lib/fastdom-promise';
-import HostedCarousel from 'commercial/modules/hosted/onward-journey-carousel';
+import { initHostedCarousel } from './onward-journey-carousel';
 
-const loadOnwardComponent = (
+const getPlaceholdersFromDom = (dom = document): Array<any> =>
+    [].slice.call(dom.getElementsByClassName('js-onward-placeholder'));
+
+const generateUrlFromConfig = (
+    c: {
+        page: {
+            ajaxUrl: string,
+            pageId: string,
+            contentType: string,
+        },
+    } = config
+): string => `${c.page.ajaxUrl}/${c.page.pageId}/${c.page.contentType.toLowerCase()}/onward.json`;
+
+const insertHTMLfromPlaceholders = (
+    json: { html: string },
+    placeholders: Array<any>
+): void => {
+    placeholders[0].insertAdjacentHTML('beforeend', json.html);
+};
+
+export const loadOnwardComponent = (
     start: () => void,
-    stop: () => void
+    stop: () => void,
+    getPlaceholders: () => Array<any> = getPlaceholdersFromDom,
+    getUrl: () => string = generateUrlFromConfig,
+    getJson: (url: string, options: RequestOptions) => Promise<any> = fetchJson,
+    insertHtmlFn: (
+        json: { html: string },
+        placeholders: Array<any>
+    ) => void = insertHTMLfromPlaceholders,
```

Comment:
This was an experiment in trying to make loadOnwardComponent a bit purer, and have less responsibilities, to see if I could: 
1) extract out some of the responsibility to new functions
2) pass those new functions in as variables
3) make it easier to write robust tests, and add more testing to commercial
4) keep it readable
5) Not hate myself when I have to come back and understand this code again in 6 months

 which is why feedback is good! If this has a higher mental load, I may have to think about it again...